pkg/fleet/installer/installer.go

```diff
@@ -995,5 +999,41 @@ func mergeConfigs(rawConfigs [][]byte) (map[string]configFile, error) {
 			}
 		}
 	}
+
+	if len(configOrder) == 0 {
+		// Early return if no config order is provided
+		return mergedFiles, nil
+	}
+
+	// Inject fleet_layers into datadog.yaml
+	datadogConfig := make(map[string]interface{})
+	datadogFile, ok := mergedFiles["/datadog.yaml"]
+	if ok {
+		err := json.Unmarshal(datadogFile.Contents, &datadogConfig)
+		if err != nil {
+			return nil, fmt.Errorf("could not unmarshal datadog.yaml contents: %w", err)
+		}
+	} else {
+		datadogFile = configFile{
+			Path:     "/datadog.yaml",
+			Action:   configFileActionAdd,
+			Contents: json.RawMessage(`{"fleet_layers": []]}`),
+		}
+	}
+
+	// Add fleet_layers configuration
+	datadogConfig["fleet_layers"] = configOrder
+	datadogConfig["config_id"] = configOrder[len(configOrder)-1]
```

Comment:
nit: i'm a bit worried that we're losing info here, maybe a simple join would be better?